spec.md

```diff
@@ -1402,6 +1408,122 @@ Applying integer decoding with a divisor of `100` to create an array of 32-bit f
 [ 1.00, 1.00, 1.00, 1.00, 0.50, 0.50 ]
 ```
 
+### Extra data
+
+The following are fields that are __not__ supplied by the *RCSB*, and are provided for applications to transfer non-molecular data along with the molecular data.
+
+The restrictions on each of the fields are __not__ enforced at decode-time and are simple conventions to help guide applications.
+
+| Name            | length-restrictions     | 
+|-----------------|-------------------------|
+| bondProperties  | length of numBonds      |
+| atomProperties  | length of numAtoms      |
+| groupProperties | length of numGroups     |
+| chainProperties | length of numChains     |
+| modelProperties | length of numModels     |
+| extraProperties | None                    |
+
+#### bondProperties
+
+*Optional field*
+
+*Type*: [Map](#types)
+
+*Description*: A field meant to store information on specific bonds.  Must have the same length as [numBonds](#numbonds).
```

Comment:
If it must be of same length as numBonds then it can be an array instead of a map, right?

Reply:
Sorry I will think of a way to rephrase this, the idea is that you have `keys` with the `values` being `arrays` with length `numBonds`  if that makes sense!

Reply:
So, a map where the keys are integers in [0, numBonds - 1].

Reply:
One note, using maps is more flexible but does not easily allow for the bespoke array compression techniques.

Reply:
it would be like:
```
for something with 5 atoms
atomProperties: {
  "colorList": ["#ff0000", "#008000", "#ff0000", "#008000", "#ff0000"],
  "representationList": [0, 0, 0, 1, 1]
}

extraProperties: {
  "availableRepresentations": {0: "lines", 1: "spheres", 2: "surface", 3: "ball and stick", 4: "cartoon"}
}
```

Reply:
availableRepresentations in this case doesn't have to be 5, while colorList and representationList do have to have len(5)

Reply:
From the examples it looks to me the type should be Array (or List) and not Map.

Reply:
Does the way it is now make more sense @arose ?